### ISSUE TITLE
Create API documentation

### ISSUE THREAD

**Issue by Empiree** [Empiree/DeftSharp.Windows.Input#3]
You can help with creating documentation.

**Comment by jaketanis** [comment]
hi @Empiree! I can help with API Documentation. Let me know if help is still needed.

**Comment by Empiree** [comment]
> hi @Empiree! I can help with API Documentation. Let me know if help is still needed.

Hey @jaketanis, that would be awesome! 

I'm currently working on documentation for Keyboard classes. You could take on one of these options:

- [MouseListener](https://github.com/Empiree/DeftSharp.Windows.Input/blob/main/DOCUMENTATION.md#mouselistener)
- [MouseManipulator](https://github.com/Empiree/DeftSharp.Windows.Input/blob/main/DOCUMENTATION.md#mousemanipulator)
- [NumpadListener](https://github.com/Empiree/DeftSharp.Windows.Input/blob/main/DOCUMENTATION.md#numpadlistener)

You can use the KeyboardListener documentation as a basis, but it's not in perfect condition yet either. So don't hesitate to try something new. I'll be glad for any help:)

**Comment by jaketanis** [comment]
Sounds great! I will get acquainted with the processes and understand how it works. If I have any questions I'll let you know! 👍

**Comment by Empiree** [comment]
> Sounds great! I will get acquainted with the processes and understand how it works. If I have any questions I'll let you know! 👍

Yes, feel free to contact me if you need any details. Thanks for your participation!

**Comment by Empiree** [comment]
Hi @jaketanis ! The documentation is mostly ready, it describes the basic things. But if you still want to help, you can edit it, format it or add something new. I have no experience in creating technical documentation, so your help would be very much appreciated. 

In the future, I plan to move all documentation to a website with navigation and a better layout.

**Comment by jaketanis** [comment]
> Hi @jaketanis ! The documentation is mostly ready, it describes the basic things. But if you still want to help, you can edit it, format it or add something new. I have no experience in creating technical documentation, so your help would be very much appreciated. 
> 
> 
> 
> In the future, I plan to move all documentation to a website with navigation and a better layout. 

Sounds good @Empiree. I'll suggest edits over time. I am not very familiar with .NET, so I am taking some time to understand the application and how the APIs work. I have been quite busy these last couple of weeks, so I am having to take it slow.

Moving the documentation to a seperate website can be done and I can help with that. If they are REST APIs, SwaggerUI is a popular method. This uses OpenAPI specs. If not, another popular method is through mkdocs. This is a static-site generator, which uses markdown as the file type. This seems like a good idea since all of the documentation is already in markdown.

In terms of best practices for technical documentation, it depends on the API. Are the APIs in this project REST APIs? If so, OpenAPI specification is the standard for documentation.

**Comment by jaketanis** [comment]
If it is not a REST API, we could use [dotnet.github.io/docfx/index.html](docfx). There is a setup for .NET APIs. 

Let me know what you think. 👍

**Comment by Empiree** [comment]
Hi @jaketanis 

I think the word API confused you and it was my mistake. The point is that the library is used locally in the system and does not provide for interaction outside the computer via remote access. Therefore, there is no such API here, nor is there a REST architecture. 

By API I meant a specification of interaction between the user and input devices such as keyboard and mouse. Since the library is built on Windows system calls using Windows API.

**Comment by jaketanis** [comment]
> Hi @jaketanis 
> 
> 
> 
> I think the word API confused you and it was my mistake. The point is that the library is used locally in the system and does not provide for interaction outside the computer via remote access. Therefore, there is no such API here, nor is there a REST architecture. 
> 
> 
> 
> By API I meant a specification of interaction between the user and input devices such as keyboard and mouse. Since the library is built on Windows system calls using Windows API. 

Ahh, I see! Thanks for the clarification @Empiree. I think that will make documentation for this project a little easier. I will look at it later tonight and provide some more questions. 😁
